### PR TITLE
Feature/apt package

### PR DESCRIPTION
**Pull request by stevendborrelli**
- adds ability to install/remove debian packages
- refactors some of the code in the package resource to be shared

### CODE REVIEW

**Comment by ryane** [comment]
Haven't figure out exactly why yet but I am getting the following error when trying to apply https://github.com/asteris-llc/converge/blob/1a41999d0730a51fb8982c3c51db5b07a4ac99de/samples/apt.hcl:

```
$ /vagrant/converge apply --local /vagrant/examples/rbenv/mc.hcl
2016-11-07T15:15:18Z |INFO| serving     addr=:47740 component=rpc
2016-11-07T15:15:18Z |WARN| skipping module verification        component=client
2016-11-07T15:15:18Z |INFO| got status  component=client file=/vagrant/examples/rbenv/mc.hcl id=root/package.apt.mc run=STARTED stage=APPLY
2016-11-07T15:15:19Z |INFO| got status  component=client file=/vagrant/examples/rbenv/mc.hcl id=root run=STARTED stage=APPLY

root/package.apt.mc:
 Error: root/package.apt.mc still has changes after apply: exit status 100
 Messages:
 Has Changes: yes
 Changes:
  mc: "absent" => "present"

Summary: 2 errors, 1 changes

 * root/package.apt.mc: root/package.apt.mc still has changes after apply: exit status 100
 * root: error in dependency "root/package.apt.mc"
```

**Comment by ryane** [comment]
ignore the above (not using sudo, duh)

File: docs/content/resources/package.apt.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+menu:
+  main:
+    parent: resources
+---
+
+
+apt Package manages system packages with `apt` and `dpkg`. It assumes that
+both `apt` and `dpkg` are installed on the system, and that the user has
+permissions to install, remove, and query packages.
+
```
Comment (ryane) [inline docs/content/resources/package.apt.md]: should we add a note about using `group` to ensure that multiple apt package installations do not run in parallel? (at least until we add an implicit group into the apt resource)

File: resource/package/apt/preparer.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	Name string `hcl:"name" required:"true" `
+
+	// State of the package. Present means the package will be installed if
+	// missing; Absent means the package will be uninstalled if present.
+	State pkg.State `hcl:"state" valid_values:"present,absent"`
+}
+
+// Prepare a new package
+func (p *Preparer) Prepare(ctx context.Context, render resource.Renderer) (resource.Task, error) {
+	if p.State == "" {
```
Comment (ryane) [inline resource/package/apt/preparer.go]: might want to validate that Name != ""

Comment (stevendborrelli) [inline resource/package/apt/preparer.go]: added

**Pull request merged**